src/framework/audio/audiotypes.h

```diff
@@ -53,31 +54,87 @@ using MixerChannelId = int32_t;
 using AudioSourceName = std::string;
 using AudioResourceId = std::string;
 using AudioResourceIdList = std::vector<AudioResourceId>;
+using AudioResourceVendor = std::string;
+
+enum class AudioResourceType {
+    Undefined = -1,
+    FluidSoundfont,
+    VstPlugin
+};
+
+struct AudioResourceMeta {
+    AudioResourceId id;
+    AudioResourceType type = AudioResourceType::Undefined;
+    AudioResourceVendor vendor;
+
+    bool hasNativeEditorSupport = false;
+
+    bool isValid() const
+    {
+        return !id.empty() && !vendor.empty();
+    }
+
+    bool operator==(const AudioResourceMeta& other) const
+    {
+        return id == other.id
+               && vendor == other.vendor
```

Comment:
What about comparing by type?

Reply:
I think I've missed that one when have been adding the type field

thanks